app/models/sample.rb

```diff
@@ -77,7 +77,11 @@ def referenced_resources
     sample_type.sample_attributes.select(&:seek_resource?).map do |sa|
       value = get_attribute_value(sa)
       type = sa.sample_attribute_type.base_type_handler.type
-      type.constantize.find_by_id(value['id']) if value && type
+      if value.kind_of?(Array)
+        value.map {|v| type.constantize.find_by_id(v['id']) if v && type} 
+      else
+        type.constantize.find_by_id(value['id']) if value && type
+      end
```

Comment:
Similar to above comment, you can force the value to an array (with `Array(v)`) to avoid duplication:
```
Array(value).map {|v| type.constantize.find_by_id(v['id']) if v && type} 
```